azure/services/async/interfaces.go

```diff
@@ -41,11 +41,18 @@ type FutureHandler interface {
 // Creator is a client that can create or update a resource asynchronously.
 type Creator interface {
 	FutureHandler
-	CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, azureautorest.FutureAPI, error)
+	CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, existingResource interface{}) (interface{}, azureautorest.FutureAPI, error)
+	Get(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, error)
```

Comment:
It's arguably odd to have `Get` in a `Creator` interface, but I'm ok with it here.